apps/astarte_realm_management/test/astarte_realm_management/device_removal/scheduler_test.exs

```diff
@@ -0,0 +1,87 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2025 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+defmodule Astarte.RealmManagement.DeviceRemoval.SchedulerTest do
+  @moduledoc """
+  Tests for the device remover scheduler.
+  """
+  alias Astarte.DataAccess.Repo
+  alias Astarte.DataAccess.Realms.Realm
+  alias Astarte.DataAccess.Device.DeletionInProgress
+  alias Ecto.UUID
+  alias Astarte.RealmManagement.DeviceRemoval.DeviceRemover
+  alias Astarte.RealmManagement.DeviceRemoval.Scheduler
+
+  use Astarte.RealmManagement.DataCase, async: true
+  use Mimic
+
+  describe "Test device remover scheduler" do
+    setup %{astarte_instance_id: astarte_instance_id} do
+      scheduler =
+        case Scheduler.start_link(wait_start: true) do
+          {:ok, scheduler} -> scheduler
+          {:error, {:already_started, scheduler}} -> scheduler
+          error -> raise "could not start the device scheduler: #{inspect(error)}"
+        end
+
+      Astarte.DataAccess.Config
+      |> allow(self(), scheduler)
+      |> stub(:astarte_instance_id, fn -> {:ok, astarte_instance_id} end)
+      |> stub(:astarte_instance_id!, fn -> astarte_instance_id end)
+
+      %{scheduler: scheduler}
+    end
+
+    test "Calls device remover in a separate supervised task", %{
+      scheduler: scheduler,
+      realm: realm
+    } do
+      device_id = UUID.bingenerate()
+
+      %DeletionInProgress{
+        device_id: device_id,
+        vmq_ack: true,
+        dup_start_ack: true,
+        dup_end_ack: true
+      }
+      |> Repo.insert!(prefix: Realm.keyspace_name(realm))
+
+      Task.Supervisor
+      |> allow(self(), scheduler)
+      |> expect(:start_child, fn Astarte.RealmManagement.DeviceRemoverSupervisor,
+                                 DeviceRemover,
+                                 :run,
+                                 [deletion_in_progress],
+                                 [restart: :transient] ->
+        assert %DeletionInProgress{
+                 device_id: ^device_id,
+                 vmq_ack: true,
+                 dup_start_ack: true,
+                 dup_end_ack: true
+               } = deletion_in_progress
+
+        :ok
+      end)
```

Comment:
As agreed upon IRL, this is the interesting part of the test, while the rest is just checking that OTP's basic functionalities work. So let's just test the query that lists devices to be deleted, and trust OTP for the rest